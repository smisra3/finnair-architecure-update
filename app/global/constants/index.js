/* TODO: Remove lint exception */
/* eslint-disable import/prefer-default-export */
export const DEVICE_TYPE = 'DEVICE_TYPE';
export const IS_TABLET = 'IS_TABLET';
export const REGION_TYPE = 'REGION_TYPE';
export const CONFIG_KEYS = 'CONFIG_KEYS';
export const CURRENT_ROUTE = 'CURRENT_ROUTE';
export const PAGE_URL = 'PAGE_URL';
export const PAGE_QUERY = 'PAGE_QUERY';
export const GLOBAL_DATA_FAILURE = 'FIL/GLOBAL/GLOBAL_DATA_FAILURE';
export const CHECK_AUTH = 'FIL/GLOBAL/CHECK_AUTH';
export const CHECK_AUTH_SUCCESS = 'FIL/GLOBAL/CHECK_AUTH_SUCCESS';
export const SET_USER_STATE = 'FIL/GLOBAL/SET_USER_STATE';
export const PAGE_ORIGIN = 'PAGE_ORIGIN';
export const LABELS_STATE_KEY = 'labels';

export const USER_STATE_LOGGED_IN = 'FIL/GLOBAL/SUCCESS';
export const USER_STATE_RECOGNIZED = 'FIL/GLOBAL/RECOGNIZED';
export const USER_STATE_ANONYMOUS = 'FIL/GLOBAL/ANONYMOUS';

export const GET_LABELS = 'FIL/GLOBAL/GET_LABELS';
export const GET_LABELS_SUCCESS = 'FIL/GLOBAL/GET_LABELS_SUCCESS';
export const GET_LABELS_FAILURE = 'FIL/GLOBAL/GET_LABELS_FAILURE';
