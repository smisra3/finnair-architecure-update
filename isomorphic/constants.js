const HEARTBEAT_SESSION_NAME = 'JSESSIONID';
const ENV_DEVELOPMENT = 'development';
const ENV_PRODUCTION = 'production';
const SERVICE_WORKER_FILE = '/service-worker.js';
const API_PROXY_PATH = '/api';
const CLEAR_SERVICES_CACHE = '/clear-services-cache';
const HEALTH_CHECK = '/healthcheck';
const PROTOCOL_HTTPS = 'https';
const RESPONSE_OK = 200;
const RESPONSE_INTERNAL_SERVER_ERROR = 500;
const STATIC_FOLDER_PREFIX = '/static/';

module.exports = {
  HEARTBEAT_SESSION_NAME,
  ENV_DEVELOPMENT,
  ENV_PRODUCTION,
  SERVICE_WORKER_FILE,
  API_PROXY_PATH,
  CLEAR_SERVICES_CACHE,
  PROTOCOL_HTTPS,
  RESPONSE_OK,
  RESPONSE_INTERNAL_SERVER_ERROR,
  STATIC_FOLDER_PREFIX,
  HEALTH_CHECK,
};
