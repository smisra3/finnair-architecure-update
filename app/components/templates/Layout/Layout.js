// @flow
import { ThemeProvider } from "styled-components";
import type { Node } from "react";
import { connect } from "react-redux";
import Head from "next/head";
import get from "lodash/get";

import Theme from "../../../styles/theme/";
import Header from "../../organisms/Header";
import { MOBILE } from "../../../constants";
import API from "../../../utils/fetch";

type Props = {
  children: Node,
  deviceType: string,
  hasRightGutter?: boolean
};

const Layout = ({ children, deviceType, hasRightGutter }: Props): Node => {
  // `deviceType` is now accessible to all component styles using `props.theme.deviceType`
  Theme.deviceType = deviceType;
  /**
   * isTabletCheckout Setting for theming at component level
   * for all checkout pages loading on tablet device.
   */
  Theme.hasRightGutter = hasRightGutter;
  API.setDeviceType(deviceType);

  const isMobile = deviceType === MOBILE;
  /**
   * check whether on checkout pages to have right Space/gutter in the window,
   * for chat Icon overlaping issue in Ipad.
   */
  const width = hasRightGutter ? 1100 : 1000;

  return (
    <ThemeProvider theme={Theme}>
      <main>
        <Head>
          {isMobile ? (
            <meta
              name="viewport"
              content="user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, width=device-width, shrink-to-fit=no"
            />
          ) : (
            <meta name="viewport" content={`width=${width}`} />
          )}
        </Head>
        <Header />
        {children}
      </main>
    </ThemeProvider>
  );
};

Layout.defaultProps = {
  hasRightGutter: false
};

/* istanbul ignore next */
const mapDispatchToProps = () => ({});

const mapStateToProps /* istanbul ignore next */ = (state): { deviceType: string } => ({
  deviceType: get(state, ["global", "globalData", "deviceType"]),
  isNavigationDrawerOpen: get(state, ["global", "header", "isNavigationDrawerOpen"]),
  topBanner: get(state, ["global", "header", "topBanner"]),
  isTablet: get(state, ["global", "globalData", "isTablet"]),
  topBannerModalContent: get(state, ["global", "header", "topBannerModalContent"])
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Layout);
export { Layout };
