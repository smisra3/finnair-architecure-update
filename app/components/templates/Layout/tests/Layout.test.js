import { shallow } from 'enzyme';
import 'jest-styled-components';
import { Layout } from '../Layout';
import LayoutElement from '../Layout.style';
import { DESKTOP, MOBILE } from '../../../../constants';

const props = {
  children: <div>HELLO</div>,
  deviceType: DESKTOP,
  isNavigationDrawerOpen: true,
  topBanner: 'TOPBANNER',
  hasRightGutter: 1100,
};

describe('Layout Component', () => {
  let layout = '';

  beforeEach(() => {
    layout = shallow(<Layout {...props} />);
  });

  test('should render correctly', () => {
    expect(layout).toMatchSnapshot();
  });

  test('should render when deviceType is Mobile', () => {
    props.deviceType = MOBILE;
    props.hasRightGutter = undefined;
    layout = shallow(<Layout {...props} />);
    expect(layout).toMatchSnapshot();
  });

  test('should render the styled component appropriately', () => {
    const styledLayout = shallow(<LayoutElement {...props} />);
    expect(styledLayout).toMatchSnapshot();
  });
});
