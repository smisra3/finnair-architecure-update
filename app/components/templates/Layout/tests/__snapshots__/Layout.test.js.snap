// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should render correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColorDisabled": "grey",
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "cautionBackgroundColor": "#fffdf3",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "deviceType": "desktop",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "hasRightGutter": 1100,
      "letterSpacing": "1px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zFilter": 100,
        "zFilterActions": 400,
        "zFilterList": 150,
        "zFlyout": 200,
        "zLoader": 1000,
        "zMenu": 800,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zPlacesAutocomplete": 1100,
        "zSticky": 700,
      },
    }
  }
>
  <main>
    <SideEffect(Head)>
      <meta
        content="width=1100"
        name="viewport"
      />
    </SideEffect(Head)>
    <Header />
    <div>
      HELLO
    </div>
    <Footer />
  </main>
</ThemeProvider>
`;

exports[`Layout Component should render the styled component appropriately 1`] = `
.c0 {
  width: 100%;
  padding: 0;
  -webkit-transition: -webkit-transform 0.1s ease-in-out;
  -webkit-transition: transform 0.1s ease-in-out;
  transition: transform 0.1s ease-in-out;
  height: 100vh;
  -webkit-transform: translate3d(calc(85% + 6px),0,0);
  -ms-transform: translate3d(calc(85% + 6px),0,0);
  transform: translate3d(calc(85% + 6px),0,0);
  position: fixed;
  overflow-x: visible;
}

@media (max-width:1200px) and (min-width:768px) {
  .c0 {
    margin-left: auto;
  }
}

<div
  className="c0"
>
  <div>
    HELLO
  </div>
</div>
`;

exports[`Layout Component should render when deviceType is Mobile 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColorDisabled": "grey",
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "cautionBackgroundColor": "#fffdf3",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "deviceType": "mobile",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "hasRightGutter": false,
      "letterSpacing": "1px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zFilter": 100,
        "zFilterActions": 400,
        "zFilterList": 150,
        "zFlyout": 200,
        "zLoader": 1000,
        "zMenu": 800,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zPlacesAutocomplete": 1100,
        "zSticky": 700,
      },
    }
  }
>
  <main>
    <SideEffect(Head)>
      <meta
        content="user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, width=device-width, shrink-to-fit=no"
        name="viewport"
      />
    </SideEffect(Head)>
    <Header />
    <div>
      HELLO
    </div>
    <Footer />
  </main>
</ThemeProvider>
`;
